### DISCUSSION TITLE
It seems like I need to reboot elastic everyday "so-elastic-restart --force"

### DISCUSSION THREAD

**Discussion by TotieBash** [Security-Onion-Solutions/securityonion#12459] [2.4]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

airgap

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

30

### Storage for /

200

### Storage for /nsm

200

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

It seems like I need to reboot elastic everyday "so-elastic-restart --force".  Otherwise SOC dashboard and Kibana are empty for the last 8-16 hours query.  If I query last 24 hours it spits out data.  When I reboot the docker by "so-elastic-restart --force" all the data in the past 16 hours are there.  Which tells me that they are being ingested by Elasticsearch but for some reason SOC dashboard and Kibana is not spitting it out.

I also have 2 other SO 2.4.40 doing the same thing.  Am the only one? 

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Are there any clues in the SOC log? `/opt/so/log/soc/sensoroni-server.log`

**Reply by TotieBash** [reply]
Thanks Chris I will check tuesday...

**Reply by TotieBash** [reply]
I'm still getting this issue on my 3 standalone POC vm.  I narrow it down to logstash log:
`OOM command not allowed when used memory > 'maxmemory'`

I also narrowed it down to `so-logstash-restart --force` being my temporary fix.  I increased my Redis maxmemory from default value of `812m` to `4096m` base on suggested link below and I will check tomorrow if that solves the problem. 
https://docs.securityonion.net/en/2.4/redis.html#queue

**Reply by cm-ops** [reply]
Sounds like Logstash issue, what are you Logstash config/settings in SOC? lsheap, batch_size, and pipeline workers?

**Reply by TotieBash** [reply]
lsheap=1000m
batch_size=125
pipeline workers=8

**Reply by cm-ops** [reply]
Try increasing your batch size to `500`, keep an eye on heap, but `1000m` should be enough. Sounds like Logstash can't keep up with what is in the Redis queue.

**Reply by TotieBash** [reply]
@cm-ops I adjusted the Logstash `pipeline_x_batch_x_size` to `500` as advised on my 3 standalone SO.  I will report back tomorrow if that solves the issue.  Thanks..

**Reply by TotieBash** [reply]
@cm-ops I don't see OOM warning in logstash anymore but it is still choking on data.  The `so-redis-count` is still backed-up.  And all 3 of them started choking data at the same time 3 hours ago.  I am using Suricata for metadata engine to extract network flows.  I am not using Zeek for all 3  systems, in fact `docker ps` does not show zeek.  Below is what is on the logstash tail logs multiple times.

`[logstash.outputs.elasticsearch] Retrying failed action ...........pipeline=>"suricata.common"`
`[logstash.outputs.elasticsearch] Retrying individual bulk actions that failed or were rejected by the previous bulk request`

**Reply by TotieBash** [reply]
Past 6hour
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/f2fe5d28-c054-4a35-962f-4850d0007242)

Past 24hours
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/03c17f0e-9dac-406d-bde7-33d330cda5ca)

log output
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/cc47c294-6c55-464d-9b4c-cb5aa00de48f)

**Reply by TotieBash** [reply]
I change logstash batch_size again from `500` to `2048`, I'll check tomorrow if it stays up.

**Reply by TotieBash** [reply]
Mar 21 still the same isssue with batch_size `2048`
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/1f8bfdf6-771a-4001-96e5-71d6beca5685)

**Reply by cm-ops** [reply]
What was the actual error in the above Logstash log? The screenshot cuts it out.

**Comment by TotieBash** [comment]
illegal_state_exception.  Here's full screenshot.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/280ea7be-1523-4b06-bc48-3ead25521e2a)

**Reply by cm-ops** [reply]
Looks like a non-existant pipeline error. What does `sudo so-elasticsearch-pipelines-list | grep suricata` return?

**Reply by TotieBash** [reply]
@cm-ops 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/7aedc6cb-b1b8-4987-b08f-d612b2b0bae3)

**Reply by TotieBash** [reply]
I change the metadata engine on 1 of the 3 standalone SO back to original Zeek.  My hypothesis is that Suricata as the metadata engine is one of the root of my issue.

**Reply by cm-ops** [reply]
Okay, I will probably have to test this and see if I can replicate it. Just so I am doing the same, 2.4.50 on standalone architecture with Suricata as metadata correct?

**Reply by TotieBash** [reply]
@cm-ops correct.  Standalone 2.4.50 vm using iso image.

I change back 1 of the 3 SO metadata engine from Suri to Zeek yesterday.  Today it works, I didn't need to so-logstash-reboot.  However I still see the "non-existent pipeline [suricata.]" at the tail of logstash.log.  So I am not sure if that log is 100% relevant...  I will monitor and see if it stays up till tomorrow...

**Reply by TotieBash** [reply]
The SO running Zeek metadata is up for 2nd day now...

**Reply by cm-ops** [reply]
I have my test server running, but have not run into this issue yet. Will continue to monitor. Do you have any Elasticsearch/Logstash customizations configured?

**Reply by TotieBash** [reply]
No customization, I do use the Cisco IOS integration plugins but other than that it is vanilla base install...  And the monitor port is only monitoring around 10Kbps average.

**Reply by cm-ops** [reply]
I added the Cisco IOS integration and still cannot replicate. But I took a closer look at your Logstash log screenshot of the error. Would you be able to check the suricata.common pipeline and see if there are errors in `{ "set":    { "field": "event.dataset",             "value": "{{ message2.event_type }}"                            } }, ` that processor? You can check it with `so-elasticsearch-pipeline-stats suricata.common`. 

I see the event_type in your log is `ike`, that set processor should be making the `event.dataset = suricata.ike`, but if you look towards the end it is just `suricata`. So it is failing on `{ "pipeline": { "if": "ctx?.event?.dataset != null", "name": "suricata.{{event.dataset}}" } }` hence the pipeline failure for `suricata.`.

Also, if you can get that piece of the log out of the `eve.json`, I can test with it. I have not been able to replicate the issue so far.

**Reply by TotieBash** [reply]
@cm-ops I didn't see any error when I ran `so-elasticsearch-pipeline-stats suricata.common`.
Where can I email the snippet of the sanitized eve.json?


The SO running Zeek metadata is up for 7 days.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/01d976d6-f983-4abb-b022-e549c430f19d)

The SO running Suri is down since last Thursday when I so-logstash-restart
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/1a28c89a-297b-461d-8308-083d9675ecb7)

The second SO running Suri is down as well since Thursday.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/86748904/43d35caa-af71-43ce-a270-a9ad1f21b4c0)

**Reply by cm-ops** [reply]
If the log is sanitized are you able to add it to this thread?

**Comment by TotieBash** [comment]
@cm-ops here is the sanitize log.
[ike-suricata-eve-log-sanitized.txt](https://github.com/Security-Onion-Solutions/securityonion/files/14876986/ike-suricata-eve-log-sanitized.txt)

**Reply by cm-ops** [reply]
Thanks, I was able to replicate, suspect it is something malformed in the json and it is failing to parse out the json. Will talk with the devs about my findings.

**Reply by TotieBash** [reply]
ok thank you so much...  In the mean time I am using Zeek as my work around....  I still desire SURI to handle metadata and I guess the future direction is for SURI to also handle full-packet-capture as well.  Since STENO has been archived by Google.

**Reply by TotieBash** [reply]
Chris just some amplifying info.  This is how I extracted the log:
`zcat eve-2024-04-02*.json.gz | grep 1583301747739198 | jq `

Then I sanitized the following:
src_ip
dest_ip
vendor_ids
Then all the obvious `REDACTED` values

**Reply by cm-ops** [reply]
I gathered that from the logs, I had to undo the jq to get them to work. When the log was proper json, I could see it in the console, when it wasn't, I saw the error for the wrong pipeline. Some intial thoughts were to add an ignore failure to the pipeline or an intermediary drop if it cannot parse the json. But I have not had time to test recently.
